src/main/java/com/shipmentEvents/handlers/EventHandler.java

```diff
@@ -60,31 +57,37 @@ public String handleRequest(ScheduledEvent scheduledEvent, Context context) {
     private void processShipmentUpdates(final LambdaLogger logger) throws InterruptedException {
 
         final List<String> bucketsToProcess = Constants.BUCKETS_TO_PROCESS;
-        final Map<String, Pair<Long, String>> latestStatusForTrackingNumber = new HashMap<String, Pair<Long, String>>();
-        final Map<String, List<KeyVersion>> filesToDelete = new HashMap<String, List<DeleteObjectsRequest.KeyVersion>>(); 
+        final Map<String, Pair<Long, String>> latestStatusForTrackingNumber = new HashMap<>();
+        final Map<String, List<ObjectIdentifier>> filesToDelete = new HashMap<>();
         for (final String bucketName : bucketsToProcess) {
-            final List<KeyVersion> filesProcessed = processEventsInBucket(bucketName, logger, latestStatusForTrackingNumber);
+            final List<ObjectIdentifier> filesProcessed = processEventsInBucket(bucketName, logger, latestStatusForTrackingNumber);
             filesToDelete.put(bucketName, filesProcessed);
         }
-        final AmazonS3 s3Client = EventHandler.getS3Client();
+        final S3Client s3Client = EventHandler.getS3Client();
 
         //Create a new file in the Constants.SUMMARY_BUCKET
         logger.log("Map of statuses -> " + latestStatusForTrackingNumber);
         String summaryUpdateName = Long.toString(System.currentTimeMillis());
         
-        EventHandler.getS3Client().putObject(Constants.SUMMARY_BUCKET, summaryUpdateName, latestStatusForTrackingNumber.toString());
+        //EventHandler.getS3Client().putObject(Constants.SUMMARY_BUCKET, summaryUpdateName, latestStatusForTrackingNumber.toString());
+        PutObjectRequest putObjectRequest = PutObjectRequest.builder().bucket(Constants.SUMMARY_BUCKET).key(summaryUpdateName).build();
+        EventHandler.getS3Client().putObject(putObjectRequest, RequestBody.fromString(latestStatusForTrackingNumber.toString()));
         
         long expirationTime = System.currentTimeMillis() + Duration.ofMinutes(1).toMillis();
-        while(System.currentTimeMillis() < expirationTime) {
-            if (s3Client.doesObjectExist(Constants.SUMMARY_BUCKET, summaryUpdateName)) {
+        while (System.currentTimeMillis() < expirationTime) {
+            HeadObjectRequest headObjectRequest = HeadObjectRequest.builder().bucket(Constants.SUMMARY_BUCKET).key(summaryUpdateName).build();
+            HeadObjectResponse headObjectResponse = s3Client.headObject(headObjectRequest);
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

This code appears to be waiting for a resource before it runs. You could use the waiters feature to help improve efficiency. Consider using ObjectExists or ObjectNotExists. For more information, see [https://aws.amazon.com/blogs/developer/waiters-in-the-aws-sdk-for-java/](https://aws.amazon.com/blogs/developer/waiters-in-the-aws-sdk-for-java/)